### ISSUE TITLE
HccePose is using BGR instead of RGB

### ISSUE THREAD

**Issue by wortelus** [WangYuLin-SEU/HCCEPose#8]
Hi,
first of all thank you for this awesome work, performance of this model is astonishing just a few hours into the training on my RTX 5060 Ti. Really enjoy working with this model and the repo workflow itself, too :). I think I found a logical bug during training the HccePose model itself (mainly the `bop_loader.py` file) , and subsequent testing using `s4_p3` files depending on `tester.py`.

The training data pipeline in `bop_loader.py` loads images via `cv2.imread()` (which returns BGR)
but never converts them to RGB before applying ImageNet normalization and feeding into the network.

This causes two issues:
1. **ImageNet normalization is applied to wrong channels** during training in the `bop_loader.py` — mean/std for R (0.485/0.229) is applied to B channel and vice versa
2. **Train/test mismatch** — `tester.py` seems to normalize RGB values on RGB ImageNet values, however swaps the channels back to BGR before feeding them to the network.

### Training
`HccePose/bop_loader.py` — both `preprocess()` methods (training and test dataset classes):
```python
# cv2.imread returns BGR, but no conversion is done
rgb_c_pil = Image.fromarray(np.uint8(rgb_c)).convert('RGB')  # does NOT swap channels
# ImageNet RGB normalization applied to BGR data
```
Fix is simple -- add `rgb_c = cv2.cvtColor(rgb_c, cv2.COLOR_BGR2RGB)` before `Image.fromarray()` in both preprocess() methods (or earlier in the pipeline).

### Testing / Inference

However the `tester.py` is somewhat more confusing in this regard, because `s4_p3` files (tests for img/video bin picking) do more confusing steps, such as:
```python
ret, frame = cap.read()
if not ret:
    break
frame = cv2.cvtColor(frame, cv2.COLOR_RGB2BGR)
```
OpenCV always reads image as BGR, so the flag is incorrect. However the result is correctly RGB, as is also normalized in the `tester.py`:
```python
composed_transforms_img = transforms.Compose([
            transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
            ])
# Correct normalization for the RGB data which were passed to the tester...
```
However the tester.py does this to swap the channels during the `.flip(dims = [1])`:
```python
# Basically, I understand it as a swap from BHWC -> BCHW, the C (dim=1) is then swapped to BGR after the RGB normalization
img_torch_hccepose = composed_transforms_img(img_torch.permute(0,3,1,2) / 255).flip(dims = [1])
```

Probably the fix should be changing the
```python
# From
frame = cv2.cvtColor(frame, cv2.COLOR_RGB2BGR)
# To (same thing functionally, but less confusing)
frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
```
and...
```python
# From
img_torch_hccepose = composed_transforms_img(img_torch.permute(0,3,1,2) / 255).flip(dims = [1])
# To
img_torch_hccepose = composed_transforms_img(img_torch.permute(0,3,1,2) / 255)
```
It seems the network is evaluated on BGR data with correct ImageNet normalization, while the training feeds the network BGR data with mistakenly swapped ImageNet normalization... I hope I have a clear image of the processing pipeline, particularly the testing one, because that one is more confusing. 

If we apply these fixes, both the training and inference pipelines will correctly feed normalized RGB data to the pretrained EfficientNet/ResNet backbones, as originally intended.

wortelus
